### PR TITLE
Fixed the roofs not opening when building aircraft units

### PR DESCRIPTION
**Pull request by Mailaender**
Closes https://github.com/OpenRA/ra2/issues/182.

### CODE REVIEW

File: OpenRA.Mods.RA2/Traits/Render/WithExitOverlay.cs [hunk 1]
```diff
@@ -62,24 +64,16 @@ void INotifyDamageStateChanged.DamageStateChanged(Actor self, AttackInfo e)
 			overlay.ReplaceAnim(RenderSprites.NormalizeSequence(overlay, e.DamageState, overlay.CurrentSequence.Name));
 		}
 
-		void INotifyProduction.UnitProduced(Actor self, Actor other, CPos exit)
+		public void UnitProduced(Actor self, Actor other, CPos exit)
```
Comment (abcdefg30) [inline OpenRA.Mods.RA2/Traits/Render/WithExitOverlay.cs]: What is this change for?

Comment (Mailaender) [reply OpenRA.Mods.RA2/Traits/Render/WithExitOverlay.cs]: That was an accident.

File: mods/ra2/rules/allied-structures.yaml [hunk 2]
```diff
@@ -397,12 +397,28 @@ gaweap:
...
+	GrantConditionOnProduction@helicopters:
+		Actors: shad
+		Condition: roof-open
+		Duration: 50
+		ShowSelectionBar: false
+	GrantConditionOnProduction@zeppelins:
+		Actors: zep
+		Condition: roof-open
+		Duration: 100
+		ShowSelectionBar: false
```
Comment (MustaphaTR) [inline mods/ra2/rules/allied-structures.yaml]: This seems to be not working but it is an upstream issue.

Comment (Mailaender) [reply mods/ra2/rules/allied-structures.yaml]: I see a pink selection bar during the roof opening.

Comment (MustaphaTR) [reply mods/ra2/rules/allied-structures.yaml]: Yes, that's what the value is meant to remove.

Comment (Mailaender) [reply mods/ra2/rules/allied-structures.yaml]: Fixed https://github.com/OpenRA/OpenRA/pull/17704.

**Comment by MustaphaTR** [comment]
I guess this can be merged regardless of that bug so doing it.

**Pull request merged**